lib/commands.js

```diff
@@ -628,6 +629,45 @@ class Commands {
             inferredConcertoJsonModel.models[0]
         );
     }
+
+    /**
+     * Decorate a given model with given list of dcs and vocab files and print the result
+     * @param {string} modelFile - the model to which vocab and decorator has to be applied
+     * @param {string[]} dcsFiles - the decorator files to be applied to model
+     * @param {string[]} vocFiles - the vocab files to be applied to model
+     * @param {object} options - optional parameters
+     * @returns {string} - Depending on the options, CTO string or JSON string
+     */
+    static async decorate(modelFile, dcsFiles,vocFiles,options) {
+        try {
+            const modelContent = fs.readFileSync(path.resolve(modelFile), 'utf-8');
+            const allDCSFiles = dcsFiles ? dcsFiles.map(file => fs.readFileSync(file, 'utf8')) : [];
+            const allVocsFiles = vocFiles ? vocFiles.map(file => fs.readFileSync(file, 'utf8')) : [];
+            let modelManager = new ModelManager();
+            modelManager.addModel(modelContent);
+            allDCSFiles.forEach(content => {
+                modelManager = DecoratorManager.decorateModels(modelManager, JSON.parse(content));
+            });
+            const vocManager = new VocabularyManager({ missingTermGenerator: VocabularyManager.englishMissingTermGenerator });
+            const namespace = modelManager.getModels().map(obj=>obj.name.replace(/\.cto$/, ''))[0];
```

Comment:
The namespace for a ModelFile should be retrieved using modelFile.getNamespace() not inferred from its file name. The file name can be anything, or might even be absent, whereas the namespace is inside the ModelFile.

Reply:
`getNamespaces()` appends concerto namespaces as well. Filtering them out felt hacky. in `getmodels()` as far as what I saw, the name was `{namespace}.cto` and only had manually added ctos.
So I used this approach.

The other thing I can do is have a namespace array and populate it as below.
`let model=modelManager.addModel(modelContent);`
`namespaces.push(model.namespace)`

Which one do yoy think makes most sense?

Reply:
Getting the namespace from the model file using `modelFile.getNamespace()` and then building a CTO file name from that is the only reliable approach imo.